test/e2e/_incluster-test-code/memcached_test.go

```diff
@@ -421,11 +422,12 @@ func memcachedOperatorMetricsTest(t *testing.T, f *framework.Framework, ctx *fra
 func getMetrics(t *testing.T, f *framework.Framework, labels map[string]string, ns, port string) ([]byte, error) {
 	// Get operator pod
 	pods := &v1.PodList{}
-	opts := &client.ListOptions{}
-	opts.InNamespace(ns)
-	opts.MatchingLabels(labels)
-	opts.MatchingField("status.phase", "Running")
-	err := f.Client.List(goctx.TODO(), pods, client.UseListOptions(opts))
+	opts := []client.ListOption{
+		client.InNamespace(ns),
+		client.MatchingLabels(labels),
+		client.MatchingField("status.phase", "Running"),
```

Comment:
[`client.MatchingField` is deprecated](https://github.com/kubernetes-sigs/controller-runtime/blob/d7467fcf9e1c5817231f2768ffd626a244f2221b/pkg/client/options.go#L316-L322). Can we use `client.MatchingFields` here instead?

Reply:
Hmm interesting, since `client.MatchingField()` returns a `client.MatchingFields`. I'll use `client.MatchingFields` directly.